cli/src/semgrep/output.py

```diff
@@ -496,6 +499,7 @@ def _build_output(
                 time=cli_timing,
                 explanations=explanations,
                 rules_by_engine=rules_by_engine,
+                # this flattens the information into just distinguishing "pro" and "not-pro"
```

Comment:
Is this because the sarif formatter also outputs a check for `interfile: true`? (Comment could include more context)

Reply:
no, it was meant to be a reference to the implementation of `to_engine_kind`, which will simply inject into `PRO`, regardless of the particular kind of pro engine variant

Reply:
Why do we choose to flatten this though?

Reply:
the type of `CliOutputExtra` requires an `engine_kind`, which is a sum of only `PRO` and `OSS`. we could make this all four variants, but as I said in the summary, there are places where this makes life hard for us. in particular, this is the same variant that we use to label our matches, which would require a fair sum of work to categorize matches as `PRO_INTRAFILE` findings or `PRO_LANGUAGE` findings, etc. this should be cleaned up later, but I don't think this is a problem for now

Reply:
I see. The comment could clarify that the output engine_kind has less information. I would be fine with merging this and improving the comment later though